### PR TITLE
SDKS-495 & SDKS-497 Implement NumberAttributeInputCallback and Boolea…

### PR DESCRIPTION
**Pull request by witrisna**
…nAttributeInputCallback

### CODE REVIEW

File: forgerock-auth/src/main/java/org/forgerock/android/auth/callback/NumberAttributeInputCallback.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+     */
+    @Keep
+    public NumberAttributeInputCallback(JSONObject jsonObject, int index) throws JSONException {
+        super(jsonObject, index);
+    }
+
+    @Override
+    protected void setAttribute(String name, Object value) {
+        super.setAttribute(name, value);
+        if ("value".equals(name)) {
```
Comment (rodrigoareis) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/callback/NumberAttributeInputCallback.java:41]: May use constant VALUE available in AbstractCallback class?

File: forgerock-auth/src/test/java/org/forgerock/android/auth/callback/NumberAttributeInputCallbackTest.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                "                {\n" +
+                "                    \"name\": \"IDToken1validateOnly\",\n" +
+                "                    \"value\": false\n" +
+                "                }\n" +
+                "            ]\n" +
+                "        }");
+        NumberAttributeInputCallback callback = new NumberAttributeInputCallback(raw, 0);
+        Assertions.assertThat(callback.getValue()).isEqualTo(123);
+    }
+}
```
Comment (rodrigoareis) [inline forgerock-auth/src/test/java/org/forgerock/android/auth/callback/NumberAttributeInputCallbackTest.java:165]: Do you think is valid to include a unit test to test a Double as input? I could not find the specification for this callback on the server side, but what is expected if a Float value is provided, should it return null as it is now? Are Double and Integer the only ones supported by this callback on the server?

File: forgerock-auth/src/test/java/org/forgerock/android/auth/callback/StringAttributeInputCallbackTest.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        Assertions.assertThat(callback.getPrompt()).isEqualTo("Last Name");
+        Assertions.assertThat(callback.isRequired()).isTrue();
+        Assertions.assertThat(callback.getPolicies().getString("name")).isEqualTo("sn");
+        Assertions.assertThat(callback.getFailedPolicies()).isEmpty();
+        Assertions.assertThat(callback.getValidateOnly()).isFalse();
+        Assertions.assertThat(callback.getValue()).isEmpty();
+
+
+    }
+}
```
Comment (rodrigoareis) [inline forgerock-auth/src/test/java/org/forgerock/android/auth/callback/StringAttributeInputCallbackTest.java:115]: Do you think is valid to include more unit tests to test the input value, such as non-empty strings, very long strings, special characters, and so on?

File: forgerock-auth/src/main/java/org/forgerock/android/auth/callback/NumberAttributeInputCallback.java (hunk 1 above)
Comment (witrisna) [reply forgerock-auth/src/main/java/org/forgerock/android/auth/callback/NumberAttributeInputCallback.java:41]: Good catch, updated.

File: forgerock-auth/src/test/java/org/forgerock/android/auth/callback/NumberAttributeInputCallbackTest.java (hunk 2 above)
Comment (witrisna) [reply forgerock-auth/src/test/java/org/forgerock/android/auth/callback/NumberAttributeInputCallbackTest.java:234]: A test case for double input is added.

File: forgerock-auth/src/test/java/org/forgerock/android/auth/callback/StringAttributeInputCallbackTest.java (hunk 3 above)
Comment (witrisna) [reply forgerock-auth/src/test/java/org/forgerock/android/auth/callback/StringAttributeInputCallbackTest.java:115]: The test more focus on parsing the input, there is no special logic on assigning the input json, it just assigned it as is.
